design/file-descriptors.md

```diff
@@ -0,0 +1,14 @@
+# File Descriptors
+File Descriptors ([witx type `fd`](https://github.com/WebAssembly/WASI/blob/master/phases/snapshot/docs.md#-fd)) are an abstract way of representing access to a resource. More often than not, they are processes (WIP), sockets (WIP), files, or even the console.
+
+# The Console
```

Comment:
As a minor style nit, when there are multiple sections, use `##` or so, as `#` is meant for titles.

Reply:
Gotcha!